pkg/controller/daemon/daemon_controller.go

```diff
@@ -1259,6 +1259,18 @@ func (dsc *DaemonSetsController) simulate(newPod *v1.Pod, node *v1.Node, ds *app
 		})
 	}
 
+	// This is memory efficient copy of metav1.IsControlledBy where we don't deep copy all references.
+	// This should reduce memory allocations as we call this for every single pod on every node for every daemonset
+	// that is not owned by this daemonset.
+	isControlledByDaemonSet := func(p *v1.Pod) bool {
```

Comment:
make it a helper func at the bottom of the file and pass uid in as well.

Reply:
fixed.